scripts/selecionar-tudo.js

```diff
@@ -7,12 +7,26 @@ function renderBotaoSelecionarTudo(customToolbar, document) {
 
 function criarBotaoSelecionarTudo() {
   const botaoSelecionarTudo = document.createElement("BUTTON");
-  botaoSelecionarTudo.classList.add('botao-selecionar-tudo');
+  botaoSelecionarTudo.id = "botao-selecionar-tudo";
+  botaoSelecionarTudo.className = "botao-selecionar-tudo";
   botaoSelecionarTudo.innerHTML = 'Selecionar Tudo';
   return botaoSelecionarTudo;
 }
 
+function toggleBotaoSelecionarTudo() {
+  const botaoSelecionarTudo = document.getElementById("botao-selecionar-tudo");
+  if(botaoSelecionarTudo.className == "botao-selecionar-tudo") {
+    botaoSelecionarTudo.className = "botao-desselecionar-tudo";
+    botaoSelecionarTudo.innerHTML = 'Desselecionar Tudo';
+  }
+  else {
+    botaoSelecionarTudo.className = "botao-selecionar-tudo";
+    botaoSelecionarTudo.innerHTML = 'Selecionar Tudo';
+  }
+}
+
 function handleSelecionarTudo(event, document) {
+  toggleBotaoSelecionarTudo();
   event.preventDefault();
```

Comment:
```suggestion
  event.preventDefault();
  toggleBotaoSelecionarTudo(event.target);
```
Aqui você já pode passar o próprio botão como parâmetro para a função, evitando assim ter que acessar a DOM novamente (pelo id).
Também é saudável deixar o "event.preventDefault()" sempre no início, para evitar o comportamento padrão do browser de submeter o formulário.

Reply:
Tem alguma recomendação entre usar ID ou classe?

Reply:
Acho que pelo ID é mais rápido